### ISSUE TITLE
WandbLogger does not log model topology

### ISSUE THREAD

**Issue by gau-nernst** [PyTorchLightning/pytorch-lightning#8638]
## 🐛 Bug

Following the documentation [here](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.loggers.wandb.html), calling `.watch(model)` from the WandbLogger object should log model topology, but currently it does not.

Inspecting the code:

https://github.com/PyTorchLightning/pytorch-lightning/blob/c99e2fe0d2bf713f35054eaa0d521ee7f6030786/pytorch_lightning/loggers/wandb.py#L202-L203

WandbLogger does not pass a value for `log_model` parameter to the wandb object, which is `False` by default.

https://github.com/wandb/client/blob/5ce4a13f44e0d47693affd9a494dae5d91d42438/wandb/sdk_py27/wandb_watch.py#L17-L19

If I directly call `.watch()` from the wandb object, model topology is logged correctly.

```python
logger = WandbLogger()
logger.experiment.watch(model, log_graph=True)
````

### To Reproduce

Colab link to the BoringModel: https://colab.research.google.com/drive/1keZ8hlWt2vgJDgK4e2IiXiN68npfsPiJ?usp=sharing

### Expected behavior

WandbLogger should either automatically log model topology (by passing `log_graph=True` to wandb) or let the user specify that by adding `log_graph` as an extra optional keyword argument.

### Environment

From `collect_env_details.py`

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 1.4.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.11
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021

**Comment by justusschock** [comment]
Hi @gau-nernst Thanks for the issue and the detailed description. Would you be interested in contributing a PR with a fix for this? :)

**Comment by gau-nernst** [comment]
Hello @justusschock,
Sure, it should be a simple fix. Do I need to write a test case for this? Seems like it's not simple to check whether the model topology is actually logged in a unit test.

**Comment by justusschock** [comment]
@gau-nernst great, thanks. I think this could be done by mocking the wandb calls. cc @awaelchli who is our mocking expert :P

**Comment by gau-nernst** [comment]
I'm not familiar with mocking, although I know the concept. Any guidance will be great!

**Comment by justusschock** [comment]
you can see the examples of existing tests (that also use mocking) here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/loggers/test_wandb.py

the overall documentation on mocking can be found here: https://docs.python.org/3/library/unittest.mock.html

Does that help already or do you have more specific questions? :)

If you encounter some issues, let us know

**Comment by awaelchli** [comment]
Sounds good. My suggestion is to just implement your fix/feature, then see if any existing tests are breaking. Once you have it on a PR we can develop a new unit test for it <3

**Comment by gau-nernst** [comment]
Hello, I have created a PR for this at #8662

Some points for discussion

- Should we test the default behavior of `WandbLogger.watch()`? (test for `log_graph=True`, (and `log="gradients"`, `log_freq=100` also?), when there are no optional arguments)
- Currently I updated a test to check if `log_graph` is passed when `WandbLogger.watch()` is called. Do we need to test for the internal function in wandb that is responsible for logging the model graph also?
- There is a reason for wandb to disable logging model graph by default but I'm not sure why (see [here](https://github.com/wandb/client/pull/2395)). I override this behavior because I think it makes more sense this way.
